lab_1_keywords_tfidf/main.py

```diff
@@ -99,12 +147,16 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(term_freq, dict) and isinstance(idf, dict) and type_of_elements(term_freq, tuple, str, float)\
+            and type_of_elements(idf, tuple, str, float) and term_freq:
+        max_idf = math.log(47 / 1)
+        tfidf = {term: term_f * idf.get(term, max_idf) for term, term_f in term_freq.items()}
+        return tfidf
+    else:
+        return None
 
 
-def calculate_expected_frequency(
-    doc_freqs: dict[str, int], corpus_freqs: dict[str, int]
-) -> Optional[dict[str, float]]:
+def calculate_expected_frequency(doc_freqs: dict[str, int], corpus_freqs: dict[str, int]) -> Optional[dict[str, float]]:
```

Comment:
corpus freqs is supposed to have integer values, change the function signature back, please

Reply:
still not fixed